File: spring-jpa/jpa_programming/jpa-basic/src/main/java/jpateam/Member.java
package jpateam;

import javax.persistence.*;

//@Entity
public class Member {
    @Id
    @GeneratedValue
    @Column(name = "MEMBER_ID")
    private Long id;

    @Column(name = "USERNAME")
    private String name;

    //@Column(name = "TEAM_ID")
    //private Long teamId;

    //Member가 many Team이 One
    @ManyToOne
    @JoinColumn(name = "TEAM_ID")
    private Team team;

    public Member() {
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Team getTeam() {
        return team;
    }

    public void changeTeam(Team team) {
        this.team = team;
        team.getMembers().add(this);
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

}

File: spring-jpa/jpa_programming/querydsl/src/main/kotlin/com/example/querydsl/entity/Member.kt
package com.example.querydsl.entity

import javax.persistence.*

@Entity
class Member(
    var username: String? = null,
    var age: Int,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "team_id")
    var team: Team? = null,

    @Id @GeneratedValue
    @Column(name = "member_id")
    var id: Long? = null
) {
    fun changeTeam(team: Team) {
        this.team = team
        team.members.add(this)
    }
}
